specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json

```diff
@@ -0,0 +1,2349 @@
+          "type": "string",
+          "description": "FQDN for the master pool."
+        },
+        "agentPoolProfiles": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedClusterAgentPoolProfile"
+          },
+          "description": "Properties of the agent pool."
+        },
+        "linuxProfile": {
+          "$ref": "#/definitions/ContainerServiceLinuxProfile",
+          "description": "Profile for Linux VMs in the container service cluster."
+        },
+        "windowsProfile": {
+          "$ref": "#/definitions/ManagedClusterWindowsProfile",
+          "description": "Profile for Windows VMs in the container service cluster."
+        },
+        "servicePrincipalProfile": {
+          "$ref": "#/definitions/ManagedClusterServicePrincipalProfile",
+          "description": "Information about a service principal identity for the cluster to use for manipulating Azure APIs."
+        },
+        "addonProfiles": {
+          "additionalProperties": {
+            "$ref": "#/definitions/ManagedClusterAddonProfile"
+          },
+          "description": "Profile of managed cluster add-on."
+        },
+        "nodeResourceGroup": {
+          "type": "string",
+          "description": "Name of the resource group containing agent pool nodes."
+        },
+        "enableRBAC": {
+          "type": "boolean",
+          "description": "Whether to enable Kubernetes Role-Based Access Control."
+        },
+        "enablePodSecurityPolicy": {
+          "type": "boolean",
+          "description": "(PREVIEW) Whether to enable Kubernetes Pod security policy."
+        },
+        "networkProfile": {
+          "$ref": "#/definitions/ContainerServiceNetworkProfile",
+          "description": "Profile of network configuration."
+        },
+        "aadProfile": {
+          "$ref": "#/definitions/ManagedClusterAADProfile",
+          "description": "Profile of Azure Active Directory configuration."
+        },
+        "privateLinkResources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedClusterPrivateLinkResource"
+          },
+          "description": "Properties related to private link cluster."
+        },
+        "apiServerAccessProfile": {
+          "$ref": "#/definitions/ManagedClusterAPIServerAccessProfile",
+          "description": "Access profile for managed cluster API server."
+        }
+      },
+      "description": "Properties of the managed cluster."
+    },
+    "ManagedClusterPrivateLinkResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified Azure resource ID."
+        },
+        "name": {
+          "type": "string",
+          "description": "The private link resource name."
+        },
+        "type": {
+          "type": "string",
+          "description": "The private link resource type."
+        },
+        "groupId": {
```

Comment:
how is groupId different from the name?

Reply:
this private link resource contract id dictated by NRP as this is a special scenario that requires joint effort from NRP/AKS RP.

I'm not expert on this feature - my understanding is that this group is a concept required by NRP to manage the private link resources.

Reply:
is this a settable property?

Reply:
how does the customer know what goes here?

Reply:
this is set by AKS RP service. Customer just read the value.

Is there any impact to ARM?

Reply:
we decided to remove this section in this api version.